package.json

```diff
@@ -11,39 +11,58 @@
     "build/**/*.js",
     "build/contracts/*",
     "build/artifacts/**/*.json",
+    "build/artifacts-ovm/**/*.json",
     "build/types/**/*.ts",
+    "build/types-ovm/**/*.ts",
     "contracts/*"
   ],
   "scripts": {
+    "prepack": "yarn setup",
     "clean": "tsc -b --clean tsconfig.json && rm -rf build/",
-    "setup": "yarn build:contracts",
-    "build:contracts": "yarn build:contracts:v0.4 && yarn build:contracts:v0.6",
-    "build:contracts:v0.4": "VERSION=v0.4 hardhat compile --show-stack-traces",
-    "build:contracts:v0.6": "VERSION=v0.6 hardhat compile --show-stack-traces",
-    "test": "hardhat test --show-stack-traces"
+    "setup": "yarn build:contracts && yarn build",
+    "lint": "echo \"Please set up linter\"",
+    "build": "tsc -p tsconfig.json",
+    "build:contracts": "yarn build:contracts:0.4 && yarn build:contracts:0.6 && yarn build:contracts:0.7 && yarn build:contracts:0.7:ovm",
+    "build:contracts:0.4": "hardhat compile --contracts 0.4 --show-stack-traces",
+    "build:contracts:0.6": "hardhat compile --contracts 0.6 --show-stack-traces",
+    "build:contracts:0.7": "hardhat compile --contracts 0.7 --show-stack-traces",
+    "build:contracts:0.7:ovm": "hardhat compile --contracts 0.7 --network optimism --show-stack-traces",
+    "test": "hardhat test --no-compile --show-stack-traces",
+    "script:oe:up": "./node_modules/@chainlink/optimism-utils/scripts/run-ci.sh",
+    "script:oe:down": "cd optimism/ops && docker-compose down -v",
+    "script:oe:clean": "cd optimism/ops && docker-compose down -v --remove-orphan --rmi all"
   },
   "dependencies": {
     "@chainlink/contracts": "^0.1.6",
-    "@openzeppelin/contracts": "^3.4.0"
+    "@chainlink/optimism-utils": "https://github.com/smartcontractkit/optimism-utils.git",
```

Comment:
Can't all of these other than OZ be moved into dev dependencies?

Reply:
"Needs to be" a proper dependency as this is now more than just a raw contracts package. It brings some useful functions for consumers like Gauntlet to use, for example, the `getContractFactory` fn to load the factory by name for a specific version & target.

Most of that code could be extracted in a separate lib, and then reused here, in Feeds Registry repo, in our core evm-contracts, etc.

The specific `"@chainlink/optimism-utils"` dependency is because of [`optimism.loadEnv` fn](https://github.com/smartcontractkit/LinkToken/blob/5d44e695db5494235695f7718654cb32c314b60f/src/optimism/index.ts#L13) which is currently repo specific and does not have much use externally.

Reply:
I've cleaned this a little, but still left the `@chainlink/optimism-utils` as a dependency. It's less confusing to use it this way currently. I agree we can organize this better, and please share your suggestions, but I guess this is good enough for now.

I vote we try to see how all of this fits into our larger set of contract repos & tools. We can try to use it (share hardhat & optimism utils) in Gauntlet, Feed Registry, core contracts, and then restructure it to fit better depending on the outcome.